pahfit/features/features.py

```diff
@@ -308,16 +308,16 @@ def _construct_table(cls, inp: dict):
         tables = []
         for (kind, features) in inp.items():
             kind_params = cls._kind_params[kind]  # All params for this kind
-            rows = []
+            rows, dtypes = [], []
             for (name, params) in features.items():
                 for missing in kind_params - params.keys():
                     if missing in cls._no_bounds:
                         params[missing] = 0.0
                     else:
                         params[missing] = value_bounds(0.0, bounds=(0.0, None))
                 rows.append(dict(name=name, **params))
-            table_columns = rows[0].keys()
-            t = cls(rows, names=table_columns)
+                dtypes.append(None if name in cls._no_bounds else cls._bounds_dtypes)
+            t = cls(rows, names=rows[0].keys(), dtype=dtypes)
```

Comment:
I changed some things. Is this the intended result:
```
rows:  [{'name': 'starlight', 'group': '_none_', 'kind': 'starlight', 'temperature': (5000, nan, nan), 'tau': (0.0, 0.0, inf)}]
names:  dict_keys(['name', 'group', 'kind', 'temperature', 'tau'])
dtype:  [dtype('<U'), dtype('<U'), dtype('<U'), dtype([('val', '<f4'), ('min', '<f4'), ('max', '<f4')]), dtype([('val', '<f4'), ('min', '<f4'), ('max', '<f4')])]
resulting t.dtype [('name', '<U9'), ('group', '<U6'), ('kind', '<U9'), ('temperature', [('val', '<f4'), ('min', '<f4'), ('max', '<f4')]), ('tau', [('val', '<f4'), ('min', '<f4'), ('max', '<f4')])]
```

Reply:
To set up this dtype, I use
```python
kind_cols = rows[0].keys()
col_dtypes = [np.dtype(type(rows[0][col_name])
                          if col_name in cls._no_bounds
                          else cls._bounds_dtype
                          for col_name in kind_cols]
```

Reply:
Something else is wrong then.  Note the Table docs:

> The values must be valid [numpy.dtype](https://numpy.org/doc/stable/reference/generated/numpy.dtype.html#numpy.dtype) initializers or [None](https://docs.python.org/3/library/constants.html#None). Any list elements with value [None](https://docs.python.org/3/library/constants.html#None) fall back to the default type.

So we should be handing a list with some `None`'s in it.  Can you try again with the variable fix above?

Reply:
I did try after doing that fix

This revealed two separate issues:
- the number of dtypes is not the same as the number of column names
names = ['name', 'group', 'kind', 'temperature', 'tau'] (5 values)
[dtype([('val', '<f4'), ('min', '<f4'), ('max', '<f4')])] (only one)
- when I manually pass `(None, None, None, dtype([('val', '<f4'), ('min', '<f4'), ('max', '<f4')]), dtype([('val', '<f4'), ('min', '<f4'), ('max', '<f4')]))`, then np.float32 is used as the default type, and the code will crash because those columns are actually strings.

Reply:
OK dumb of me.  `name` is the name of the feature, not a parameter name.  I sent up a fix which should work, and now just hard-code `str` for the type of `_no_bounds` things.  I don't love that, because we might add a thing without (possible) bounds that is not a string, but OK for now.

Reply:
More issues:
- `kind_params` does not include the generic columns 'name', 'group' and 'kind'. So the loop `dtypes = [str if x in cls._no_bounds else cls._bounds_dtype for x in kind_params]` misses those extra strings.
- `no_bounds` does not include 'kind', so this logic would give it the wrong type.